device/CMakeLists.txt

```diff
@@ -15,18 +17,82 @@ set(UMD_DEVICE_SRCS
     wormhole/wormhole_implementation.cpp
     pcie/pci_device.cpp
 )
-add_library(umd_device SHARED ${UMD_DEVICE_SRCS})
-target_link_libraries(umd_device 
-    PUBLIC yaml-cpp::yaml-cpp umd_common_directories nng uv compiler_flags
-    PRIVATE hwloc rt Boost::interprocess fmt
+
+add_library(device SHARED ${UMD_DEVICE_SRCS})
+add_library(${PROJECT_NAME}::device ALIAS device)
+add_library(${PROJECT_NAME}_device ALIAS device) # For legacy I guess
+
+target_include_directories(
+    device
+    PUBLIC
+        $<BUILD_INTERFACE:${PROJECT_SOURCE_DIR}>
+        $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}>
+        $<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}/${PROJECT_NAME}>
+        $<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}/${PROJECT_NAME}/device>
+)
+
+# flatbuffers is public - exposed to tt_metal by tt_simulation_device_generated.h
+# nng is public - exposed to tt_metal by tt_simulation_host.hpp
+target_link_libraries(
+    device
+    PUBLIC
+        nng
+        flatbuffers
+    PRIVATE
+        hwloc
+        rt
+        Boost::interprocess
+        fmt::fmt-header-only
+        uv
+        yaml-cpp::yaml-cpp
+)
+
+install(
+    TARGETS
+        device
+    EXPORT ${PROJECT_NAME}Targets
+    ARCHIVE
+        DESTINATION ${CMAKE_INSTALL_LIBDIR}
+    LIBRARY
+        DESTINATION ${CMAKE_INSTALL_LIBDIR}
+        COMPONENT dev
+)
+
+install(
```

Comment:
can we switch now to cmake install in CI which builds tests? I couldn't make this work, and then I ended up manually listing all the files required to a tarfile. 
Not saying this has to be part of this PR, but create an issue if it is left to be done.

Reply:
I filed an issue. This will take some time, because I don't think we have defined that any of the tests should be installed have we?

Reply:
Yeah, I guess our tests can also have an install command, right?